index.bs

```diff
@@ -1149,9 +1197,63 @@ The [=new=] {{MLGraphBuilder}} constructor steps are:
+    </div>
+    1. Let |descriptor| be the first argument.
+    1. If |descriptor| is not an an object that [=implements=] {{MLOperandDescriptor}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
+        1. If the [=byte length=] of |descriptor| is not supported by the underlying platform, then throw a "{{DataError}}" {{DOMException}} and stop.
+        1. If the [=check dimensions=] steps given |descriptor|.{{MLOperandDescriptor/type}} and |descriptor|.{{MLOperandDescriptor/dimensions}} return `false`, throw a "{{DataError}}" {{DOMException}} and stop.
+    1. Let |bufferView| be the second argument.
+        1. If invoking <a>validate buffer with descriptor</a> given |bufferView| and |descriptor| return `false`, then throw a "{{TypeError}}" {{DOMException}} and stop.
+    1. Let |operand| be the result of invoking the <a>create MLOperand</a> steps with [=this=], `"constant"` and |descriptor|.
+        1. If that throws, re-throw the exception and stop.
+    1. Let |bytes| be the result of invoking [[=get a copy of the bytes held by the buffer source=]] given |bufferView|.
+    1. Make a request to the underlying platform to register |operand| as a constant with |bytes| as value and store a reference to the corresponding [=implementation-defined=] object to |operand|.{{MLOperand/[[operand]]}}.
+        1. If that fails, throw an "{{OperationError}}" {{DOMException}} and stop.
+    1. Return |operand|.
+</div>
+
+#### The {{MLGraphBuilder/constant(value, type)}} method  #### {#api-mlgraphbuilder-constant-value-type}
+<div class="note">
+    **Arguments:**
+        - *value*: a number
+        - *type*: an optional {{MLOperandType}}, by default *"float32"*.
+    **Returns:**: an {{MLOperand}} object.
+</div>
+<div algorithm=constant-value-type>
+    The {{MLGraphBuilder/constant(value, type)}} steps are:
+    <div class="note">
+        The permissions and context validity have been checked by [[#api-mlgraphbuilder-constructor]] steps.
+    </div>
+    1. Let |value| be the first argument.
+        1. If |value| is not a number, then  throw a "{{TypeError}}" {{DOMException}} and stop.
+    1. Let |type| be the second argument.
+        1. If |type| is `undefined`, let |type| be `"float32"`.
+        1. Otherwise, if |type| is not one of {{MLOperandType}}, then  throw a "{{TypeError}}" {{DOMException}} and stop.
+    1. Let |descriptor| be a new {{MLOperandDescriptor}}.
+        1. Set |descriptor|.{{MLOperandDescriptor/type}} to |type|.
+        1. Set |descriptor|.{{MLOperandDescriptor/dimensions}} to `[ 1 ]`.
+    1. Let |bufferView| be an {{ArrayBuffer}} of type corresponding to |type| according to [this table](#appendices-mloperandtype-arraybufferview-compatibility), containing |value| as its single element.
+    1. Let |operand| be the result of invoking the {{MLGraphBuilder/constant(descriptor, bufferView)}} steps given |descriptor| and |bufferView|.
```

Comment:
Should it request the underlying platform to register a corresponding platform scalar operand with `|type|` and `|value|`? 

Some native ML APIs support to create scalar operand, e.g. [NNAPI](https://developer.android.com/ndk/reference/group/neural-networks#group___neural_networks_1gaf06d1affd33f3bc698d0c04eceb23298). `MLGraphBuilder.constant(value, type)` should not just be polyfilled by `MLGraphBuilder.constant(descriptor, bufferView)`.

Reply:
They require the `descriptor`, but the steps do have the gap of checking the descriptor. Before fixing either of these, we should agree (or I should understand) how they are supposed to work when `dimensions` and/or `type` is missing.